### PR TITLE
feat: BigTable online store

### PR DESCRIPTION
**Pull request by chhabrakadabra**
**What this PR does / why we need it**:

BigTable online-store implementation.

**Which issue(s) this PR fixes**:

Fixes #

# Remaining tasks

- [x] Write documentation for the BigTable online store
- [x] Update the implementation to be in line with recommendations from Google
- [x] Clean up the implementation and complete all the TODOs in here
- [x] Integrate this with the Feast test suite
- [x] Rebuild all the requirements files

### CODE REVIEW

**Review by achals — COMMENTED** [review]
looks good! Just a couple of early comments. Happy to help with testing setup for this

File: sdk/python/feast/infra/online_stores/contrib/bigtable_online_store/bigtable.py
Comment (chhabrakadabra) [inline sdk/python/feast/infra/online_stores/contrib/bigtable_online_store/bigtable.py:86]: Note to self: if row is None, then the rest of the code below needs to be bypassed using `continue`.

File: setup.py
Comment (chhabrakadabra) [inline setup.py:120]: I blame `black` 😄

File: docs/specs/online_store_format.md
Comment (adchia) [inline docs/specs/online_store_format.md:113]: seems like the default below is actually 2?

File: docs/reference/online-stores/bigtable.md
Comment (adchia) [inline docs/reference/online-stores/bigtable.md:49]: i believe this is possible? Bigtable should store multiple versions of the same key, and then you fetch the latest version at read time.

**Review by adchia — APPROVED** [review]

File: docs/specs/online_store_format.md
Comment (chhabrakadabra) [reply docs/specs/online_store_format.md:113]: The `max_versions` is set to `2`, which I think means one current version and 1 historical version. Or that's what it seems based on [this document](https://cloud.google.com/bigtable/docs/samples/bigtable-create-family-gc-max-versions). Quoted from the document linked above:

```python
    print("Creating column family cf2 with max versions GC rule...")
    # Create a column family with GC policy : most recent N versions
    # where 1 = most recent version

    # Define the GC policy to retain only the most recent 2 versions
    max_versions_rule = column_family.MaxVersionsGCRule(2)

    column_family2 = table.column_family("cf2", max_versions_rule)
    column_family2.create()
    print("Created column family cf2 with Max Versions GC Rule.")
```

File: docs/reference/online-stores/bigtable.md
Comment (chhabrakadabra) [reply docs/reference/online-stores/bigtable.md:49]: Yes, that's a really good point. I'll fix it right away. Thanks.

**Comment by chhabrakadabra** [comment]
@adchia I forgot to add the signature in the last commit. I guess I'll have to ask you to lgtm this once more please. 🙏

**Pull request merged**
